### DISCUSSION TITLE
How to bundle x86 and M1 libraries to a single M1 executable

### DISCUSSION THREAD

**Discussion by yogeshsofficial** [pyinstaller/pyinstaller#7656] [Help]
Lets say, I have 10 dependencies(dylibs) of which 2 are from x86 architecture and rest from M1. I dont have the source code of the x86 libraries since its a third party and I have list support.

When I try to package using pyinstaller 4.5 in Mac M1 I get an asserstion error which is obvious stating that there is an architecture mismatch for the x86 library.

Is there any way to bundle both the architectures library in to 1 executable using pyinstaller? 

Thanks in advance.

**Comment by rokm** [comment]
I think in v4.10, we relaxed the architecture compatibility check so that it raises errors only on incompatible python extensions but not shared libraries (because an extension is loaded within the process, and hence needs to be compatible; but if extension is a multi-arch binary, its slices might actually be linked against single-arch dylibs...). So you should try with more recent version of PyInstaller.
